extensions/agroal/deployment/src/test/java/io/quarkus/agroal/test/XaRequiresXaDatasourceTest.java

```diff
@@ -0,0 +1,31 @@
+package io.quarkus.agroal.test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import javax.enterprise.inject.spi.DeploymentException;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.wildfly.common.Assert;
+
+import io.quarkus.test.QuarkusUnitTest;
+
+public class XaRequiresXaDatasourceTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addAsResource("application-wrongdriverkind-datasource.properties", "application.properties"))
+            .assertException(t -> {
+                assertEquals(DeploymentException.class, t.getClass());
+            });
+
+    @Test
+    public void xaRequiresJta() {
+        //Should not be reached: verify
+        Assert.assertTrue(false);
```

Comment:
This uses `org.wildfly.common.Assert`, probably not what you're looking for. Maybe rather `org.junit.jupiter.api.Assertions#fail(java.lang.String)`?

Reply:
Interesting. Got that as I copied another test as starting point.. a quick grep reveals that the WildFly assert is being used a lot all over :/

Reply:
OK, let's fix that in another PR, I need this one to go through CI or I won't be able to backport it.

Reply:
Sure, that's unrelated. I was just wondering how this could happen :)